ballerina/tests/test.bal

```diff
@@ -0,0 +1,262 @@
+    io:println("Message Details: ", res);
+    test:assertEquals(res.id, messageId, msg = "Retrieved message ID does not match the requested ID");
+
+}
+
+@test:Config {dependsOn: [testCreateThread]}
+function testCreateMessage() returns error? {
+    if (threadId == "") {
+        test:assertFail(msg = "No thread ID available. Ensure testCreateThread runs first.");
+    }
+
+    CreateMessageRequest createMsgReq = {
+        role: "user",
+        content: "Can you help me solve the equation `3x + 11 = 14`?"
+    };
+
+    MessageObject res = check openAIAssistant->/threads/[threadId]/messages.post(createMsgReq, headers);
+    io:println("Created Message: ", res);
+    test:assertNotEquals(res.id, "");
+    messageId = res.id;
+
+}
+
+@test:Config {dependsOn: [testCreateThread]}
+function testListMessages() returns error? {
+    if (threadId == "") {
+        test:assertFail(msg = "No thread ID available. Ensure thread creation test runs first.");
+    }
+
+    ListMessagesResponse res = check openAIAssistant->/threads/[threadId]/messages.get(headers);
+    io:println("ListMessagesResponse: ", res);
+    test:assertTrue(res is ListMessagesResponse);
+}
+
+@test:Config {dependsOn: [testCreateThread, testCreateAssistant]}
+function testCreateRun() returns error? {
+    CreateRunRequest runReq = {
+        assistant_id: assistantId,
+        model: "gpt-3.5-turbo",
+        instructions: "You are a personal math tutor. Assist the user with their math questions.",
+        temperature: 0.7,
+        top_p: 0.9,
+        max_prompt_tokens: 400,
+        max_completion_tokens: 200
+    };
+
+    RunObject resp = check openAIAssistant->/threads/[threadId]/runs.post(runReq, headers);
+    io:println("Created Run: ", resp);
+    test:assertNotEquals(resp.id, "", msg = "Run creation failed: No Run ID returned");
+    runId = resp.id;
+}
+
+@test:Config {dependsOn: [testCreateThread]}
+function testListRuns() returns error? {
+    ListRunsResponse res = check openAIAssistant->/threads/[threadId]/runs.get(headers);
+    io:println("Runs in Thread: ", res.data);
+    test:assertNotEquals(res.data.length(), 0, msg = "No runs found in the thread");
+
+}
+
+@test:Config {dependsOn: [testCreateAssistant]}
+function testCreateThreadAndRun() returns error? {
+    // CreateThreadAndRunRequest createThreadAndRunReq = {
+    //     assistant_id: assistantId,
+    //     model: "gpt-3.5-turbo",
+    //     instructions: "You are a personal math tutor. Assist the user with their math questions.",
+    //     temperature: 0.7,
+    //     top_p: 0.9,
+    //     max_prompt_tokens: 400,
+    //     max_completion_tokens: 200
+    // };
+
+    // RunObject resp = check openAIAssistant->/threads/runs.post(createThreadAndRunReq, headers);
+    // io:println("Created Thread and Run: ", resp);
+    // test:assertNotEquals(resp.id, "", msg = "Thread and Run creation failed: No Run ID returned");
+
+}
+
+@test:Config {}
```

Comment:
Please updates all the test cases with the proper test groups (`live_server` and `mock_server`)